src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -220,6 +226,22 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
         }
     }
 
+    private void validateDataTypes(List<ColumnModel> columns, Errors errors) {
+        List<ColumnModel> invalidColumns = new ArrayList<>();
+        for (ColumnModel column : columns) {
+            // spring defaults user input not belonging to the TableDataType enum to null
+            if (column.getDatatype() == null) {
+                invalidColumns.add(column);
+            }
+        }
+        if (invalidColumns.size() > 0) {
```

Comment:
I think `isEmpty()` is more idiomatic here
```suggestion
        if (!invalidColumns.isEmpty()) {
```